### PR TITLE
puppet: refactored to use CmdRunner

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Refactored the module to use `CmdRunner`.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/modules/puppet.py

### CODE REVIEW

File: plugins/module_utils/puppet.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+import os
+
+from ansible_collections.community.general.plugins.module_utils.cmd_runner import CmdRunner, cmd_runner_fmt
+
+
+_puppet_path_prefix = ["/opt/puppetlabs/bin"]
```
Comment (felixfontein) [inline plugins/module_utils/puppet.py:15]: I would use upper case for file-level constants:
```suggestion
_PUPPET_PATH_PREFIX = ["/opt/puppetlabs/bin"]
```

Comment (russoz) [reply plugins/module_utils/puppet.py]: done

**Review by felixfontein — APPROVED** [review]
Assuming you tested this somehow, I think it should be fine.

Merging in ~a week if nobody complains.

**Comment by russoz** [comment]
I actually wrote a unit test - it can/should be largely expanded in the future. I will take this week to do just that.

**Comment by felixfontein** [comment]
Did the test pass before the refactoring?

**Comment by russoz** [comment]
No, especially because the unit test is checking for `[puppet, agent, ...]` whilst the previous code was working with one single string `"puppet agent ..."`.

**Comment by felixfontein** [comment]
@russoz should I merge now, or still wait?

**Comment by russoz** [comment]
I could not do much on the testing, so maybe wait a little bit longer.

**Comment by russoz** [comment]
I screwed up the rebase in this one. Closing, will raise another PR.

**Pull request closed without merging**